bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Scheduler.java

```diff
@@ -68,73 +67,75 @@ public static Strand getStrand() {
         return strandHolder.get().strand;
     }
     public Object callFunction(Module module, String functionName, StrandMetadata metadata, Object... args) {
-        Strand parentStrand = Scheduler.getStrand();
-        if (parentStrand != null) {
-            boolean runnable = parentStrand.isRunnable();
-            if (!runnable) {
-                parentStrand.resume();
-            }
-            ValueCreatorAndFunctionType functionType = getGetValueCreatorAndFunctionType(module, functionName);
-            Object[] argsWithDefaultValues = getArgsWithDefaultValues(functionType.valueCreator(),
-                    functionType.functionType(), parentStrand, args);
-            Object result = functionType.valueCreator().call(parentStrand, functionName, argsWithDefaultValues);
-            if (!runnable) {
-                parentStrand.yield();
-            }
-            return result;
+        Strand strand = Scheduler.getStrand();
```

Comment:
Since the logic in `callFunction` and `callMethod`, shall we refactor the common parts to a function and reuse them?

Reply:
I will send a separate PR with refactoring.